src/main/kotlin/no/nav/klage/oppgave/service/OppgaveService.kt

```diff
@@ -0,0 +1,43 @@
+package no.nav.klage.oppgave.service
+
+import no.nav.klage.oppgave.clients.FETCH_LIMIT
+import no.nav.klage.oppgave.clients.OppgaveClient
+import no.nav.klage.oppgave.domain.Oppgave
+import org.springframework.stereotype.Service
+
+@Service
+class OppgaveService(
+        private val oppgaveClient: OppgaveClient,
+        private val hjemmelParsingService: HjemmelParsingService
+) {
+    fun bulkUpdateHjemmel() {
+        var offset = 0
+
+        var oppgaveResponse = oppgaveClient.fetchHjemmel(offset)
+
+        while (oppgaveResponse.antallTreffTotalt > 0) {
+            val oppgaverWithNewHjemmel = setHjemmel(oppgaveResponse.oppgaver)
+
+            oppgaverWithNewHjemmel.forEach { oppgaveClient.putOppgave(it) }
+
+            offset += FETCH_LIMIT
+
+            oppgaveResponse = oppgaveClient.fetchHjemmel(offset)
```

Comment:
Finnes jo et par mulige strategier for å gjøre dette.

Slike jeg leser din kode nå:

1. Hente alle oppgaver uten hjemmel. Blir f.eks. totalt 2500 stk. Men kun fra 0 - 10 blir hentet.
2. Hvis 2500 > 0, så oppdater de og lagre tilbake i «oppgave-db»
3. Øke offset til 10.
4. Hente alle oppgaver uten hjemmel. Blir da f.eks. 2490. Men jobber nå med de fra 10 - 20.  D.v.s. 0 - 10 blir aldri oppdatert. Vil vel bare forskyve seg videre.

Noe blir feil her tror jeg. I denne løsningen hadde man ikke trengt å endre offset, fordi man kunne alltid jobbe fra index/offset 0 på et nytt resultatsett.

Alternativt kunne man hentet ut alle 2500 først, og så lagret etterpå i egen loop.